File: app/src/main/java/com/callmangement/utils/SessionManager.java
package com.callmangement.utils;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;

public class SessionManager extends Application {
    private final SharedPreferences pref;
    private final SharedPreferences.Editor editor;
    private final Context _context;
    private static final String PREF_NAME = "ePDS";
    private static final String PERMISSION_GRANTED_BACKGROUND = "PERMISSION_GRANTED_BACKGROUND";
    private static final String PERMISSION_GRANTED = "PERMISSION_GRANTED";

    public SessionManager(Context context) {
        this._context = context;
        pref = _context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        editor = pref.edit();
    }

    public void setPermissionGrantedBackground(boolean _PERMISSION_GRANTED_BACKGROUND) {
        editor.putBoolean(PERMISSION_GRANTED_BACKGROUND, _PERMISSION_GRANTED_BACKGROUND);
        editor.apply();
    }

    public boolean getPermissionGranted() {
        return pref.getBoolean(PERMISSION_GRANTED, false);
    }

    public void setPermissionGranted(boolean _PERMISSION_GRANTED) {
        editor.putBoolean(PERMISSION_GRANTED, _PERMISSION_GRANTED);
        editor.apply();
    }

    public boolean getPermissionGrantedBackground() {
        return pref.getBoolean(PERMISSION_GRANTED_BACKGROUND, false);
    }

    public void clear() {
        editor.clear();
        editor.commit();
    }
}

File: app/src/main/java/com/callmangement/utils/SessionManager.kt
package com.callmangement.utils

import android.app.Application
import android.content.Context
import android.content.SharedPreferences

class SessionManager(private val _context: Context) : Application() {
    private val pref: SharedPreferences
    private val editor: SharedPreferences.Editor

    init {
        pref = _context.getSharedPreferences(PREF_NAME, MODE_PRIVATE)
        editor = pref.edit()
    }

    var permissionGranted: Boolean
        get() = pref.getBoolean(
            PERMISSION_GRANTED,
            false
        )
        set(_PERMISSION_GRANTED) {
            editor.putBoolean(
                PERMISSION_GRANTED,
                _PERMISSION_GRANTED
            )
            editor.apply()
        }

    var permissionGrantedBackground: Boolean
        get() {
            return pref.getBoolean(
                PERMISSION_GRANTED_BACKGROUND,
                false
            )
        }
        set(_PERMISSION_GRANTED_BACKGROUND) {
            editor.putBoolean(
                PERMISSION_GRANTED_BACKGROUND,
                _PERMISSION_GRANTED_BACKGROUND
            )
            editor.apply()
        }

    fun clear() {
        editor.clear()
        editor.commit()
    }

    companion object {
        private const val PREF_NAME: String = "ePDS"
        private const val PERMISSION_GRANTED_BACKGROUND: String = "PERMISSION_GRANTED_BACKGROUND"
        private const val PERMISSION_GRANTED: String = "PERMISSION_GRANTED"
    }
}
